src/SFML/Window/EglContext.cpp

```diff
@@ -287,67 +297,100 @@ EGLConfig EglContext::getBestConfig(EGLDisplay display, unsigned int bitsPerPixe
 {
     EglContextImpl::ensureInit();
 
-    // Set our video settings constraint
-    const EGLint attributes[] =
-        {EGL_BUFFER_SIZE,
-         static_cast<EGLint>(bitsPerPixel),
-         EGL_DEPTH_SIZE,
-         static_cast<EGLint>(settings.depthBits),
-         EGL_STENCIL_SIZE,
-         static_cast<EGLint>(settings.stencilBits),
-         EGL_SAMPLE_BUFFERS,
-         settings.antialiasingLevel ? 1 : 0,
-         EGL_SAMPLES,
-         static_cast<EGLint>(settings.antialiasingLevel),
-         EGL_SURFACE_TYPE,
-         EGL_WINDOW_BIT | EGL_PBUFFER_BIT,
-         EGL_RENDERABLE_TYPE,
-         EGL_OPENGL_ES_BIT,
-         EGL_NONE};
-
-    EGLint    configCount;
-    EGLConfig configs[1];
-
-    // Ask EGL for the best config matching our video settings
-    eglCheck(eglChooseConfig(display, attributes, configs, 1, &configCount));
-
-    // TODO: This should check EGL_CONFORMANT and pick the first conformant configuration.
-
-    return configs[0];
+    // Determine the number of available configs
+    EGLint configCount;
+    eglCheck(eglGetConfigs(display, nullptr, 0, &configCount));
+
+    // Retrieve the list of available configs
+    auto configs = std::make_unique<EGLConfig[]>(static_cast<std::size_t>(configCount));
+
+    eglCheck(eglGetConfigs(display, configs.get(), configCount, &configCount));
+
+    // Evaluate all the returned configs, and pick the best one
+    int       bestScore = 0x7FFFFFFF;
+    EGLConfig bestConfig{};
+
+    for (std::size_t i = 0; i < static_cast<std::size_t>(configCount); ++i)
+    {
+        // Check mandatory attributes
+        int surfaceType;
+        int renderableType;
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_SURFACE_TYPE, &surfaceType));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_RENDERABLE_TYPE, &renderableType));
+        if (!(surfaceType & (EGL_WINDOW_BIT | EGL_PBUFFER_BIT)) || !(renderableType & EGL_OPENGL_ES_BIT))
+            continue;
+
+        // Extract the components of the current config
+        int red;
+        int green;
+        int blue;
+        int alpha;
+        int depth;
+        int stencil;
+        int multiSampling;
+        int samples;
+        int caveat;
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_RED_SIZE, &red));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_GREEN_SIZE, &green));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_BLUE_SIZE, &blue));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_ALPHA_SIZE, &alpha));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_DEPTH_SIZE, &depth));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_STENCIL_SIZE, &stencil));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_SAMPLE_BUFFERS, &multiSampling));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_SAMPLES, &samples));
+        eglCheck(eglGetConfigAttrib(display, configs[i], EGL_CONFIG_CAVEAT, &caveat));
```

Comment:
Is it worth abstraction this `eglGetConfigAttrib` call into a lambda that we can call with the `EGL_` macro as an argument thus allowing all these local vars to be `const` and reduce code duplication or is that overkill?

```cpp
const auto getConfigAttrib = [display, configs, i](auto idkWhatVarNameToUseHere) {
    int attribute;
    eglCheck(eglGetConfigAttrib(display, configs[i], idkWhatVarNameToUseHere, &attribute));
    return attribute;
}

const int red = getConfigAttrib(EGL_RED_SIZE);
```

Reply:
This code was intentionally written to be identical in structure to the code in Glx and Wgl contexts for consistency reasons. We can consider refactoring them all at the same time as you propose in a separate issue later.